judgments/views.py

```diff
@@ -67,6 +67,45 @@ def detail(request, judgment_uri):
     return HttpResponse(template.render({"xml": judgment_xml}, request))
 
 
+def advanced_search(request):
+    params = request.GET
+    query = params.get("query")
+    court = params.get("court")
+    judge = params.get("judge")
+    party = params.get("party")
+    order = params.get("order")
+    date_from = params.get("from")
+    date_to = params.get("to")
+    page = params.get("page", 1)
+    context = {}
+    try:
+        results = api_client.advanced_search(
+            q=query,
+            court=court,
+            judge=judge,
+            party=party,
+            page=page,
+            order=order,
+            date_from=date_from,
+            date_to=date_to,
+        )
+        multipart_data = decoder.MultipartDecoder.from_response(results)
+        model = SearchResults.create_from_string(multipart_data.parts[0].text)
+
+        context["search_results"] = [
+            SearchResult.create_from_node(result) for result in model.results
+        ]
+        context["total"] = model.total
+        context["paginator"] = paginator(int(page), model.total)
+        context[
+            "query_string"
+        ] = f'query={str(query or "")}&court={str(court or "")}&party={str(party or "")}&judge={str(judge or "")}'
```

Comment:
This is a minor comment and can come as a later refactor, but to make this a bit easier to extend I'd probably put the various parameters into a dict at the top, like this:
```python
search_params = {
  "query": params.get("query"),
  "court": params.get("court"),
  "party": params.get("party"),
  "judge": params.get("judge") 
}
```

Then we can combine `join` and Python's list comprehension to build out the query string:

```python
context["query_string"] = "&".join([f'{key}={query_params[key]}' for key in query_params if query_params[key] is not None])
```

That way, when we want to add new search parameters we can just add them to the `query_params` dict.


I don't think this needs changing now, just wanted to leave this comment so we can refer back to it later if & when we're doing some refactoring